File: field-force-python/board/mapobject.py
class MapObject:
    def __init__(self, identifier, x, y):
        self.identifier = identifier
        self.x = x
        self.y = y

    def set_pos(self, x, y):
        self.x = x
        self.y = y

    def destroy(self, gamestate):
        gamestate.gamefield.field_chars[self.x][self.y] = '0'
        gamestate.gamefield.map[(self.x, self.y)].content = MapObject('0', self.x, self.y)

File: field-force-cpp/field-force-cpp/board/mapobject.cpp
#include "mapobject.h"

#include "../game/gamestate.h"

MapObject::MapObject() { }

MapObject::MapObject(char _id, int _x_pos, int _y_pos)
{
	id = _id;
	x_pos = _x_pos;
	y_pos = _y_pos;
}

void MapObject::setPos(int _x_pos, int _y_pos)
{
	x_pos = _x_pos;
	y_pos = _y_pos;
}

void MapObject::destroy(GameState* game_state)
{
	game_state->current_field.field_chars[x_pos][y_pos] = '0';
	game_state->current_field.field[x_pos][y_pos].content = MapObject('0', x_pos, y_pos);
}